### PR TITLE
Add Rigelsys/Astrogene1000 nFrame rotator driver

### PR DESCRIPTION
**Pull request by astrogene1000**

### CODE REVIEW

File: drivers/rotator/nframe.cpp [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+        // Read these values before defining focuser interface properties
+        defineProperty(&RotatorAbsPosNP);
+        defineProperty(&SettingNP);
+        defineProperty(&FocusSpeedNP);
+        defineProperty(&GotoRotatorNP);
+        loadConfig(true, SettingNP.name);
+        readPosition();
+        readSpeedInfo();
+        IDSetNumber(&RotatorAbsPosNP, nullptr);
+        IDSetNumber(&GotoRotatorNP, nullptr);
```
Comment (knro) [inline drivers/rotator/nframe.cpp:122]: is there a reason this is define before INDI::Rotator ?

File: drivers/rotator/nframe.h [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        INumberVectorProperty FocusRelPosNP;
+        INumber FocusRelPosN[1];
+
+        // Absolute Focuser position is 0 to this maximum limit. By Default, it is set to 200,000.
+        INumberVectorProperty FocusMaxPosNP;
+        INumber FocusMaxPosN[1];
+
+        // Sync
+        INumberVectorProperty FocusSyncNP;
+        INumber FocusSyncN[1];
```
Comment (knro) [inline drivers/rotator/nframe.h:94]: Why do we have focus properties?

File: drivers/rotator/nframe.h [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        ///////////////////////////////////////////////////////////////////////////////
+        // Peripheral Ports
+        ISwitchVectorProperty PerPortSP;
+        ISwitch PerPortS[3];
+        enum { PORT_MAIN, PORT_EXP, PORT_THIRD };
+
+        // Motor wiring
+        ISwitchVectorProperty WiringSP;
+        ISwitch WiringS[4];
+        enum { WIRING_LUNATICO_NORMAL, WIRING_LUNATICO_REVERSED, WIRING_RFMOONLITE_NORMAL, WIRING_RFMOONLITE_REVERSED };
```
Comment (knro) [inline drivers/rotator/nframe.h:128]: Are these applicable in the driver?

File: drivers/rotator/nframe.cpp [hunk 4]
```diff
@@ -0,0 +1,709 @@
...
+    // Stepping Phase
+    IUFillNumber(&SteppingPhaseN[0], "PHASES", "Wiring", "%.f", 0, 2, 1, 0);
+    IUFillNumberVector(&SteppingPhaseNP, SteppingPhaseN, 1, getDeviceName(), "STEPPING_PHASE", "Phase",
+                       STEPPING_TAB, IP_RW, 0, IPS_OK);
+
+
+    IUFillNumber(&FocusSpeedN[0], "FOCUS_SPEED_VALUE", "Step Rate", "%3.0f", 0.0, 255.0, 1.0, 255.0);
+    //    IUFillNumberVector(&FocusSpeedNP, FocusSpeedN, 1, getDeviceName(), "FOCUS_SPEED", "Speed", MAIN_CONTROL_TAB, IP_RW, 60, IPS_OK);
+    IUFillNumberVector(&FocusSpeedNP, FocusSpeedN, 1, getDeviceName(), "FOCUS_SPEED", "Speed", MAIN_CONTROL_TAB, IP_RW, 0,
+                       IPS_OK);
```
Comment (knro) [inline drivers/rotator/nframe.cpp:64]: What for?

File: drivers.xml [hunk 5]
```diff
@@ -311,7 +311,11 @@
             <driver name="Rigelsys NStep">indi_nstep_focus</driver>
             <version>1.2</version>
         </device>
-          <device label="RBFocus" manufacturer="RBFocus">
+	<device label="Rigelsys nFrame" manufacturer="Rigel Systems">
+            <driver name="Rigelsys nFrame">indi_nstep_rotator</driver>
```
Comment (knro) [inline drivers.xml:315]: Shouldn't this be **indi_nframe_rotator** ?

**Review by knro — CHANGES_REQUESTED** [review]
Seems like a lot of functions are used from focuser drivers and are probably not needed here. Also CMakelists.txt change was not included in the PR.

File: drivers/rotator/nframe.cpp (hunk 4 above)
Comment (astrogene1000) [reply drivers/rotator/nframe.cpp:64]: These are the current stepper rate, just copied over from the nstep focus driver, will max into Rotator nomenclature.

File: drivers.xml (hunk 5 above)
Comment (astrogene1000) [reply drivers.xml:315]: I'll clean all of it up! Totally hosed the drivers.xml and missed the CMakelists.txt

**Pull request closed without merging**